### PR TITLE
Adds Mailman gimmick job, Adds exclusive station trait system

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* Adds a mailman as a gimmick job
* Mailman only robed spell (requires mailman clothes) - they get a scroll to learn it on roundspawn
* Refactors spell cloth check
* excluisve station trait system. all of exclusive station traits roll to appear regardless of being positive or negative.
* Adds an exclusive station trait that spawns mailman instead of adding a gimmick spawn maint.  (they do not have spawnpoints. they'll spawn in arrival shuttle instead)
* and some minor changes

for their spell, it glues your mail on your hand and will unglue when your mail recipient is near you. also, 80% chance to track the direction correctly, but 20% chance to misdirect.

## Why It's Good For The Game
a job that needs to interact with people actively might be good

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image

image

image

image
image

image

image

image

</details>

## Changelog
:cl:
add: Mailman gimmick job
add: Mailman's mail tracking spell - it needs mailman's roundstarting cloths (hat, jumpsuit, laceup shoes) as the spell is actually a robed spell and it needs those clothes.
refactor: casting cloth check is now based on bitflag rather than checking everything for TRUE value
code: small tidy work for gimmick code
add: new station trait category - exclusive traits: each of exclusive trait will appear by its own chance regardless of the number of traits to appear
add: new station exclusive trait - special jobs. adds a new slot for specific jobs (currently for only mailman)
add: spells.dm define
imageadd: unused wizard.dmi images have names now (folded scrolls)
imageadd: very identical cargo card image to mailman (need to change later)
del: deprecated crew manifest related global variables
/:cl:

### CODE REVIEW

File: code/datums/station_traits/exclusive_trait.dm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+	< Excluisve station traits >
+		exclusive station traits are chosen by their own chance regardless of the amount of station trait number
+		weight is not shared among itselves. 5 means 5% chance to appear per round.
+		if there are 3 traits and these have weight 100, all of them will appear.
+*/
+/datum/station_trait/special_jobs
+	name = "Special Jobs"
+	trait_type = STATION_TRAIT_EXCLUSIVE
+	weight = 100
```
Comment (EvilDragonfiend) [inline code/datums/station_traits/exclusive_trait.dm:12]: currently set to 100% chance to appear since this might need testmerges.

**Comment by bluezorua** [comment]
delivering mail is cargos job, you really shouldn't remove a part of cargo(which heavily lacks entertaining mechanics) and add it to some gimmick job

**Comment by EvilDragonfiend** [comment]
> delivering mail is cargos job, you really shouldn't remove a part of cargo(which heavily lacks entertaining mechanics) and add it to some gimmick job

that's why it's gimmick

**Comment by LodedDiper69** [comment]
I think the main argument against this is the potential for the mailman spell to be used for validhunting. But I will be cautiously optimistic here, we can just try and see if this actually happens.
It's certainly better than b*rber, that's for sure.

**Comment by EvilDragonfiend** [comment]
> I think the main argument against this is the potential for the mailman spell to be used for validhunting. But I will be cautiously optimistic here, we can just try and see if this actually happens. It's certainly better than b*rber, that's for sure.

Adding temporary pacifist trauma upon spell use might work.

File: code/modules/jobs/job_types/gimmick.dm [hunk 2]
```diff
@@ -152,3 +160,141 @@
...
+					"arrival pod", "brig", "bridge", "medbay central", "cargo bay",
+					"research division", "somewhere hallway", "centcom", "lavaland mining office",
+					"spirit realm", "bearspace", "nullspace", "hyperspace", "voidspace",
+					"laghter demon's belly", "lizardpeople reptilian conspiracy association"
+				)
+				location = prob(90) ? "[dir2text(get_dir(usr, prob(90) ? pick(GLOB.player_list - user) : recipient_body))]" : pick(fake_locations)
+				// 81% (90% * 90%): tracks a proper person
+				// 9% (90% * 10%): tracks a wrong person
+				// 10%: shows improper location
+			to_chat(user,"<span class='notice'>Hmm... [lowertext(location)]?</span>")
```
Comment (CydiaLamiales) [inline code/modules/jobs/job_types/gimmick.dm]: I feel like this could use more 'random' text. It could be "Maybe...", "Looks like...", "I think it's..."

Comment (EvilDragonfiend) [reply code/modules/jobs/job_types/gimmick.dm]: updated.

**Comment by EvilDragonfiend** [comment]
now failed tracking has more silly messages.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
The ability to track random users and see if they are dead or not seems like it could cause troubles, but other than that it is kind of like the other gimmick jobs where they don't really need to exist.

**Comment by EvilDragonfiend** [comment]
> but other than that it is kind of like the other gimmick jobs where they don't really need to exist.

fair point to be honest

**Pull request closed without merging**